test/testdrive/tpch.td

```diff
@@ -52,20 +52,1447 @@ true
+                      AND l_shipdate >= DATE '1995-01-01'
+                      AND l_shipdate < DATE '1995-01-01' + INTERVAL '1' year
+              )
+      )
+      AND s_nationkey = n_nationkey
+      AND n_name = 'CANADA'
+  ORDER BY
+      s_name;
+1 values hashing to c33785f534c9d1a0abc37e05972a624d
+
+# Query 21
+> SELECT
+      s_name,
+      count(*) AS numwait
+  FROM
+      supplier,
+      lineitem l1,
+      orders,
+      nation
+  WHERE
+      s_suppkey = l1.l_suppkey
+      AND o_orderkey = l1.l_orderkey
+      AND o_orderstatus = 'F'
+      AND l1.l_receiptdate > l1.l_commitdate
+      AND EXISTS (
+          SELECT
+              *
+          FROM
+              lineitem l2
+          WHERE
+              l2.l_orderkey = l1.l_orderkey
+              AND l2.l_suppkey <> l1.l_suppkey
+      )
+      AND not EXISTS (
+          SELECT
+              *
+          FROM
+              lineitem l3
+          WHERE
+              l3.l_orderkey = l1.l_orderkey
+              AND l3.l_suppkey <> l1.l_suppkey
+              AND l3.l_receiptdate > l3.l_commitdate
+      )
+      AND s_nationkey = n_nationkey
+      AND n_name = 'SAUDI ARABIA'
+  GROUP BY
+      s_name
+  ORDER BY
+      numwait DESC,
+      s_name;
+2 values hashing to 9a281f0b2da0c7a03a0b0f447c420733
+
+# Query 22
+> SELECT
+      cntrycode,
+      count(*) AS numcust,
+      sum(c_acctbal) AS totacctbal
+  FROM
+      (
+          SELECT
+              substring(c_phone, 1, 2) AS cntrycode, c_acctbal
+          FROM
+              customer
+          WHERE
+              substring(c_phone, 1, 2)
+              IN ('13', '31', '23', '29', '30', '18', '17')
+              AND c_acctbal
+                  > (
+                          SELECT
+                              avg(c_acctbal)
+                          FROM
+                              customer
+                          WHERE
+                              c_acctbal > 0.00
+                              AND substring(c_phone, 1, 2)
+                                  IN (
+                                          '13',
+                                          '31',
+                                          '23',
+                                          '29',
+                                          '30',
+                                          '18',
+                                          '17'
+                                      )
+                      )
+              AND NOT
+                      EXISTS(
+                          SELECT
+                              *
+                          FROM
+                              orders
+                          WHERE
+                              o_custkey = c_custkey
+                      )
+      )
+          AS custsale
+  GROUP BY
+      cntrycode
+  ORDER BY
+      cntrycode;
+7 values hashing to 3782c67124c71b5bcb3460934dec46de
+
+################################################
+# With indexes
```

Comment:
What's the benefit of this test when we add indexes?

Reply:
The plans with and without indexes have differences in mostly the joins: The plans without indexes use Differential joins, but many of the plans with indexes use Delta joins. Also, filters get pushed down or lifted around joins differently if they are Differential vs Delta. So I was thinking that maybe we can potentially catch more errors if we run both variations.

Reply:
Ok, so we should expect the hashes to exactly match with and without indexes? SGTM.

Reply:
Yes, it was a copy-paste for both the queries and the hashes, so the hashes are matching.